drivers/SmartThings/zigbee-window-treatment/src/test/test_zigbee_window_aqara_curtain.lua

```diff
@@ -0,0 +1,346 @@
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.partially_open())
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShadeLevel.shadeLevel(25))
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "WindowShade open cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+        {
+          mock_device.id,
+          { capability = "windowShade", component = "main", command = "open", args = {} }
+        }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.closed())
+    )
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      WindowCovering.server.commands.GoToLiftPercentage(mock_device, 100)
+    })
+    test.wait_for_events()
+  end
+)
+
+test.register_coroutine_test(
+  "WindowShade close cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+        {
+          mock_device.id,
+          { capability = "windowShade", component = "main", command = "close", args = {} }
+        }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.closed())
+    )
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      WindowCovering.server.commands.GoToLiftPercentage(mock_device, 0)
+    })
+    test.wait_for_events()
+  end
+)
+
+test.register_coroutine_test(
+  "WindowShade pause cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+        {
+          mock_device.id,
+          { capability = "windowShade", component = "main", command = "pause", args = {} }
+        }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.closed())
+    )
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      WindowCovering.server.commands.Stop(mock_device)
+    })
+    test.wait_for_events()
+  end
+)
+
+test.register_coroutine_test(
+  "Handle opencloseDirection in infochanged",
+  function()
+    test.socket.zigbee:__queue_receive(
+      {
+        mock_device.id,
+        WindowCovering.attributes.CurrentPositionLiftPercentage:build_test_attr_report(mock_device, 100)
+      }
+    )
+    test.socket.capability:__set_channel_ordering("relaxed")
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.open())
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShadeLevel.shadeLevel(100))
+    )
+    test.wait_for_events()
+    test.socket.environment_update:__queue_receive({ "zigbee", { hub_zigbee_id = base64.encode(zigbee_test_utils.mock_hub_eui) } })
+
+    local updates = {
+      preferences = {
+      }
+    }
+    updates.preferences["stse.opencloseDirection"] = true
```

Comment:
having a boolean for something named `opencloseDirection` is not very clear.  I think it would be better named as something like `reverseOpenCloseDirection` if the value is going to be a boolean.

Reply:
I updated.